transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/MessageStore.kt

```diff
@@ -117,6 +121,12 @@ internal class MessageStore(
             }
         }
     }
+
+    private fun mergeInitialCustomAttributesWith(customAttributes: Map<String, String>) {
```

Comment:
You have touched a couple of important points here, so I will. try to answer them accordingly. 


1. To remove custom attribute you need to send it with value set to empty string. We do not accept nullable CA values. 
2. The idea behind keeping initial CA and updating them with new values (where new values actually override old ones) is to handle the `newChat` scenario in a proper manner. 

Given this are initial CA that were set in `createMessagingClient`: 
mapOf("name" to "John", "location" to "USA", "foodPreference" to "Salad")
Autostart request will contain all of this data.

if later on developer decided to update "location" and remove the "foodPreference", this will be the way to do that.
sendMessage("text message here", mapOf( "location" to "Canada", "foodPreference" to ""))

so send message request will have this CA:

Channel(Metadata(customAttributes = mapOf( "location" to "Canada", "foodPreference" to "")))
Note we would not send the "name" because it was not set in sendMessage. 
However the initialCustomAttributes map will look like following:

mapOf("name" to "John", "location" to "Canada", "foodPreference" to "")

Now the ConversationDisconnect feature comes into the picture.
So if session was ended by the Agent and ConversationDisconnect is set to ReadOnly, there is an option to start  a new chat using TransportSDK api `startNewChat`. 
This api will ensure that all devices that share the same token will close existing session. And entirely fresh  new session will be configured on the device that called `startNewChat`.
If Autostart is configured to be enabled Autostart request will be dispatched as well. So if we would not update the initialCustomAttributes with values that were send later, this autostart will contain data that was set initially:
 mapOf("name" to "John", "location" to "USA", "foodPreference" to "Salad")
 The idea behind caching and updating initialCustomAttributes is to send an updated CA along with autostart. Something that looks like that:
 mapOf("name" to "John", "location" to "Canada", "foodPreference" to "")


I feel like we might want to have a short call to discuss the details of this decision.

Reply:
yes, I guess we should. I don't understand why you don't just use the same set of attributes and updates it according to the sendMessage CA if given.